deployment/experiment-deployment/docker-compose.yml

```diff
@@ -0,0 +1,41 @@
+version: '3.7'
+
+services:
+  landing:
+    image: ghcr.io/xai-demonstrator/landing-page-service:latest
+    depends_on:
+      - country
+    expose:
+      - 80
+    environment:
+      - VIRTUAL_HOST=gtc.xaidemo.de
+      - VIRTUAL_PORT=80
+      - LETSENCRYPT_HOST=gtc.xaidemo.de
+      - LETSENCRYPT_EMAIL=xai.demonstrator@gmail.com
+    restart: always
```

Comment:
Ich würde hier denke ich einfach nur das GTC-Frontend auf einem NGINX hosten und nicht den ganzen Landing-Page-Service deployen.

Also in einem Unterordner von `experiment-deployment` eine eigene Dockerfile und nginx.conf anlegen und hier einbinden. Das hat auch den Vorteil, dass wir dann ggf. mehr als eine Instanz des Backend-Services laufen lassen können.

Reply:
Ah okay, also nicht die docker-compose die aktuell im /experiment-deployment/proxy Ordner ist manuell ausführen?

Reply:
Ah ne got it, zusätzlich zu dem Proxy einen NGINX auf dem dann GTC liegt?

Reply:
Genau, also einen eigenen NGINX, aber das hast du glaube ich schon richtig gelöst.

Du müsstest das Frontend auch nicht in dem Docker Image bauen, sondern du könntest es wie im prod-deployment-Workflow machen: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/.github/workflows/prod-deployment.yml#L35

D.h. du baust in GitHub Actions das Frontend, verschiebst es in `experiment-deployment/nginx` und dann kann eine `experiment-deployment/nginx/Dockerfile` sich das Frontend "lokal" holen, ohne dass es Probleme mit dem Build-Context gibt. Macht das Sinn?

Der einzige Grund, warum wir das bei der Landing Page und den Use Cases "in" Docker bauen, ist, damit man das lokal ausführen kann, ohne, dass man ein korrekt konfiguriertes Node Setup braucht. Aber wenn wir ausschließlich auf GitHub bauen, haben wir dieses Problem nicht.

Reply:
Lass uns das so machen, das ist am einfachsten und fast genau so, wie wir es für das prod-deployment lösen, da klappt das prima. (Da haben wir halt kein NGINX, aber wir kopieren da auch das Frontend einfach zu Firebase rüber.)

Reply:
Also ich bin mir nicht ganz sicher, ob ich es richtig verstanden habe. Ich hab jetzt mal was gepusht das können wir vielleicht später diskutieren. Allgemein weiß ich noch nicht ganz wie ich die Workflows so richtig testen kann. 
Aber in der Theorie ist mein Plan: Baue Frontend in Worklow, lade es in den Ordner deployment-experiment/gtc-nginx/frontend, baue dann den NGINX docker und kopiere das bereits gebaute Frontend rein

Reply:
Also es hat soweit mal manuell geklappt die docker-compose auszuführen. 
Leider wird mir nur kein Frontend angezeigt, da hab ich wohl noch eine Fehler im NGNIX.